### ISSUE TITLE
Consider sending canonical URL via HTTP header

### ISSUE THREAD

**Issue by rgaudin** [kiwix/libkiwix#1161]
In situations where kiwix-serve is serving on the www with ZIMs that are close to their source content, it acts like some kind of _mirror_.
This can have unpleasant consequences for the actual source website in that it blurs ownership to search engines and other parties.

The general solution to this is for the _copy_ entries to indicate the _source_ using the [`rel=canonical`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel#canonical) mechanism.

Should we have https://github.com/openzim/libzim/issues/904, maybe via https://github.com/openzim/libzim/issues/316, we would be in a good position to add this _Canonical URL_ information.

In order to send it without altering the ZIMs, we could use the HTTP header method, by sending

```http
Link: <https://www.source.tld/entry.html>; rel="canonical"
```

At the moment, we have no per-entry metatada but we frequently have a [ZIM-level `Source` metadata](https://wiki.openzim.org/wiki/Metadata).

We _could_ send the aforementioned HTTP header with the `Source` value (if present obviously) on every ZIM entry. It would be _abusing_ the purpose of the *canonical URL* in that it would not be the actual entry's cannonical document but at least it would indicate to search engines that kiwix-serve-served one is not.

Thoughts?

---

- https://developers.google.com/search/blog/2013/04/5-common-mistakes-with-relcanonical
- https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls#rel-canonical-header-method

**Comment by kelson42** [comment]
I agree with the proposal but the implementation of metadata at article level is not something which is scheduled to come soon. Butbwe could/should implement the HTTP header entry like proposed (when possible).
